### PR TITLE
Update fields in Payroll CSV export file

### PR DESCRIPTION
**Pull request by padv2010**
## Trello Card Link

https://trello.com/c/NMAhMvxu/128-service-settings-download-csv-for-payroll

## Description

As per requirements, we must generate the same fields as the default CSV for Payroll is expecting.

### CODE REVIEW

File: app/models/reports/payroll.rb [hunk 1]
```diff
@@ -19,16 +19,25 @@ def csv
...
+      ).map do |row|
+        Array.new(34, nil).tap do |array|
+          array[2] = row[0]
+          array[4] = row[1]
+          array[6] = row[2]
+          array[10] = row[3]
+          array[11] = row[4]
+          array[12] = row[5]
+          array[13] = row[6]
+          array[18] = "United Kingdom"
```
Comment (raul-gracia) [inline app/models/reports/payroll.rb:38]: This is very very confusing, there is no way what content is in `row[5]` and what header `array[12]` is supposed to be.

File: spec/models/reports/payroll_spec.rb [hunk 2]
```diff
@@ -41,29 +41,79 @@ module Reports
...
+          "", # PAYMENT_METHOD
+          "", # PAYMENT_FREQUENCY
+          "", # BANK_NAME
+          "", # SORT_CODE
+          "", # ACCOUNT_NUMBER
+          "", # ROLL_NUMBER
+          "", # SCHEME_AMOUNT
+          "", # PAYMENT_ID
+          "", # CLAIM_POLICIES
+          "", # RIGHT_TO_WORK_CONFIRM_STATUS
```
Comment (raul-gracia) [inline spec/models/reports/payroll_spec.rb:77]: I will add these comments to the `payroll.rb` file to make it at least more understandable what is going into what column.

**Review by raul-gracia — APPROVED** [review]

File: app/models/reports/payroll.rb (hunk 1 above)
Comment (padv2010) [reply app/models/reports/payroll.rb:38]: To understand those array indexes you just need to go to the unit test. 

On the unit test is where documentation for CSV exports happens. On the tests you can see clearly:

1. Header
2. Context and indexes

I can't see a better way to document it, but happy to make it clearer if you can suggest a better way forward.

When it comes to the `header`, I am not sure what you mean as it is not assigned. It is tested in the code below where you can see exactly the order of the columns and the indexes.

```ruby
it "returns the header in CSV format" do
  expected_header = %w[
    No.
    TITLE
    FORENAME
    FORENAME2
    SURNAME
    SS_NO
    GENDER
    MARITAL_STATUS
    START_DATE
    END_DATE
    BIRTH_DATE
    EMAIL
    ADDR_LINE_1
    ADDR_LINE_2
    ADDR_LINE_3
    ADDR_LINE_4
    ADDR_LINE_5
    ADDR_LINE_6
    ADDRESS_COUNTRY
    TAX_CODE
    TAX_BASIS
    NI_CATEGORY
    CON_STU_LOAN_I
    PLAN_TYPE
    PAYMENT_METHOD
    PAYMENT_FREQUENCY
    BANK_NAME
    SORT_CODE
    ACCOUNT_NUMBER
    ROLL_NUMBER
    SCHEME_AMOUNT
    PAYMENT_ID
    CLAIM_POLICIES
    RIGHT_TO_WORK_CONFIRM_STATUS
  ].join(",")
  
  expect(report.csv).to include(expected_header)
  end
```

This is the test for the content:

```ruby
 it "returns the data in CSV format" do
  application = create(:application, application_progress: progress)
  
  expect(report.csv).to include([
    "", # No.
    "", # TITLE
    application.applicant.given_name, # FORENAME
    "", # FORENAME2
    application.applicant.family_name, # SURNAME
    "", # SS_NO
    application.applicant.sex, # GENDER
    "", # MARITAL_STATUS
    "", # START_DATE
    "", # END_DATE
    application.applicant.date_of_birth, # BIRTH_DATE
    application.applicant.email_address, # EMAIL
    application.applicant.address.address_line_1, # ADDR_LINE_1
    application.applicant.address.address_line_2, # ADDR_LINE_2
    "", # ADDR_LINE_3
    "", # ADDR_LINE_4
    "", # ADDR_LINE_5
    "", # ADDR_LINE_6
    "United Kingdom", # ADDRESS_COUNTRY
    "", # TAX_CODE
    "", # TAX_BASIS
    "", # NI_CATEGORY
    "", # CON_STU_LOAN_I
    "", # PLAN_TYPE
    "", # PAYMENT_METHOD
    "", # PAYMENT_FREQUENCY
    "", # BANK_NAME
    "", # SORT_CODE
    "", # ACCOUNT_NUMBER
    "", # ROLL_NUMBER
    "", # SCHEME_AMOUNT
    "", # PAYMENT_ID
    "", # CLAIM_POLICIES
    "", # RIGHT_TO_WORK_CONFIRM_STATUS
  ].join(","))
  end
```

File: spec/models/reports/payroll_spec.rb (hunk 2 above)
Comment (padv2010) [reply spec/models/reports/payroll_spec.rb:77]: I am not comfortable duplicating the documentation as I can't see the benefit for it. I will wait for the [suggestion on clarifying improvements on the documentation](https://github.com/DFE-Digital/get-a-teacher-relocation-payment/pull/161#discussion_r1274796265) before taking any action, as I will probably be missing something.

**Pull request merged**